### ISSUE TITLE
Broken in macOS Catalina

### ISSUE THREAD

**Issue by maltsev** [iansinnott/alfred-maestro#25]
[It seems](https://www.alfredforum.com/topic/3447-alfred-maestro-keyboard-maestro-integration-for-alfred/?do=findComment&comment=70923) that this workflow doesn't work in macOS Catalina anymore.

@iansinnott, @nikitavoloboev can you check that, please? I don't plan to upgrade my OS version yet since Catalina doesn't support 32-bit apps.

**Comment by nikitavoloboev** [comment]
I don't play to upgrade Catalina due to its insane security popups & [sVim](https://github.com/flippidippi/sVim) not working.

**Comment by iansinnott** [comment]
@maltsev I also have no plans of upgrading. I don't generally do macOS platform dev so I don't have a spare box lying around to upgrade instead of my main computer :/ 

Sorry, for now the recommendation is do not upgrade. Obviously we'll probably all have to upgrade sooner or later but my hope is that they will sort things out with Catalina by then.

---

As an aside, Catalina seems like such a rushed, unstable, ill-conceived release on apples part. Even if it was stable, those security popups mentioned by nikitavoloboev offer such poor UX. It's troubling. Historically this is out of character for them, but the latest iOS release wasn't great either. I'm worried for the platform.

**Comment by nikitavoloboev** [comment]
https://github.com/deanishe/alfred-sublime-text/issues/20#issuecomment-539578934 

Above solution may fix the problem.

**Comment by deanishe** [comment]
I've [added the workaround to the AwGo wiki](https://github.com/deanishe/awgo/wiki/Catalina).

It currently says the same as the comment @nikitavoloboev linked. I will update it if/when a simpler solution is found.

**Comment by iansinnott** [comment]
**Update:** I now have a computer running Catalina (10.15.1) and so far everything is working as expected. I didn't have to do any additional setup either. Perhaps the latest public Catalina build addressed some issues. 

I've only been up and running with the new OS for about 30 minutes though, so take this with a grain of salt.

**Comment by dacasine** [comment]
Works for me.

What I did to avoid the notarization was to open the package once installed, simply double-click on alfred-maestre, authorize the app and execute once.

After that, everything is working fine.

**Comment by deanishe** [comment]
> I didn't have to do any additional setup either.

I believe Catalina automatically allows binaries you've compiled yourself.

**Comment by iansinnott** [comment]
> I believe Catalina automatically allows binaries you've compiled yourself.

Fair point. It's been long enough now that I don't remember if I compiled the workflow myself when I installed it, but it's possible that's the case.

**Comment by lukaskawerau** [comment]
The way to make this work:

1. Go to Alfred > Workflows
2. Select the workflow, right-click, "Open in Finder"
3. Right-click the binary file, click "Open"
4. Apple will ask you to confirm, since it can't verify there's no malware in this
5. Done

**Comment by jrasanen** [comment]
I can confirm that method Lukas described works. You only have to do this once.
 
<img width="780" alt="Screenshot 2019-11-29 at 17 06 47" src="https://user-images.githubusercontent.com/13303/69877427-dbe2f780-12ca-11ea-9dd7-d9f8265775f4.png">

**Comment by slyfogh** [comment]
It does not work for me. After clicking Allow away and restarting Alfred (via quit) right away, it brings the security warning again. Over and over again.

**Comment by slyfogh** [comment]
> The way to make this work:
> 
> 1. Go to Alfred > Workflows
> 2. Select the workflow, right-click, "Open in Finder"
> 3. Right-click the binary file, click "Open"
> 4. Apple will ask you to confirm, since it can't verify there's no malware in this
> 5. Done

Thank you. This worked.

**Comment by gridwalk** [comment]
Note- if the binary file is missing after installing the workflow you can change the file name of AlfredMaestro.alfredworkflow to AlfredMaestro.zip, unzip it and you will find the binary in there. Copy the binary into the folder described in lukaskawerau's working solution and follow those steps.

**Comment by lkuty** [comment]
Working flawlessly in 10.15.6. I installed from sources.

**Comment by deanishe** [comment]
> I installed from sources.

Yeah, if you build it yourself, Catalina won't complain. The problem is that it has a nasty habit of simply deleting unsigned binaries you downloaded.

**Comment by youtextme** [comment]
When an Alfred workflow is stuck on “Broken in macOS Catalina”: dual-hotkey is the honest trial — keep Alfred on ⌘Space, put free Raycast on ⌥Space for a week. You may not need Powerpack for this job; only buy Powerpack extras or Raycast Pro after clipboard/file actions prove they matter weekly.

https://raycast.com/?via=girish
Affiliate — I earn a commission if you upgrade to Pro.
